src/Components/Bookmark.jsx

```diff
@@ -0,0 +1,150 @@
+				<p>{Math.round(forecast.temperature.maximum)}&#8451;</p>
+				<p>{Math.round(forecast.temperature.minimum)}&#8451;</p>
+      </div>
+      </>
+  )
+}
+
+function Bookmark({x, weather, city, dailyforecast}) {
+	const days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+	return (
+		<>
+		<div className="bookmark">
+			<div className="bookmark-country-name">
+				<h1>{city}</h1>
+			</div>
+			<div className="bookmark-cloud-type">
+				<div className='weather-icon'>
+					<img src={"http://openweathermap.org/img/wn/"+dailyforecast[0].icon+"@2x.png"} alt='cloudy_img'/>
+        </div>
+				<p>{dailyforecast[0].condition}</p>
+			</div>
+			<div className="bookmark-wind-humidity">
+				<ul>
+					<li> Wind {dailyforecast[0].windspeed}m/s </li>
+					<li> Humidity {dailyforecast[0].humidity}% </li>
+				</ul>
+			</div>
+			<div className="bookmark-temp">
+				<h1>{ Math.round(dailyforecast[0].current_temp) }&#8451;</h1>
+			</div>
+			<div className="flex bookmark-daily-forecast">
+		    {dailyforecast.map((forecast, index) => (
+				  <DailyForecast key={index} day={days[forecast.date.getDay()]} forecast={forecast} />
+        ))}
+			</div>
+		</div>
+		</>
+	)
+}
+
+
+
+
+
+function BookmarksContainer({bookmarks}){
+	const [error, setError] = useState(null);
+	const [isLoaded, setIsLoaded] = useState(false);
+	const [results, setResults] = useState(null);
+	const [forecast, setForecast] = useState(null);
+	const [generic, setGeneric] = useState("app");
+	const [notfound, setFlag] = useState(false);
+	const [weatherData, setWeatherData] = useState([]);
+
+	const fetchWeather = (city) => {
+		const url = "https://api.openweathermap.org/data/2.5/forecast?q=" +
+        city +
+        "&units=metric" +
+        "&appid=" +
+        process.env.REACT_APP_APIKEY;
+		return fetch(url)
+		  .then((res) => res.json())
+		  .then((result) => {
+			
+			if (result.cod !== "200") {
+	
+			  setIsLoaded(false);
+			  if(result.name !== null){
+				setIsLoaded(true);
+				console.log("name",result.name);
+				setResults(results);
+			  }
+			  if (result["cod"] == "404") {
+				setIsLoaded(true);
+				setFlag(true);
+			  }
+			  return null;
+			}
+	
+			let hourlyForecast = [];
+			result.list.forEach((fc) => {
+			  hourlyForecast.push({
+				current_temp: fc.main.temp,
+				condition: fc.weather[0].description,
+				date: new Date(fc.dt * 1000),
+				feels_like: fc.main.feels_like,
+				temperature: {
+				  minimum: fc.main.temp_min,
+				  maximum: fc.main.temp_max,
+				},
+				icon: fc.weather[0].icon,
+				windspeed: fc.wind.speed,
+				humidity: fc.main.humidity,
+			  });
+			});
+			document.body.classList = result.list[0].weather[0].main;
+			setWeatherData(prev=>{
+				return [...prev, {forecast:hourlyForecast,city:city}]
+			});
+			setIsLoaded(true);
+		  })
+		  .catch((error) => {
+			setIsLoaded(true);
+			setError(error);
+		  });
+	  };
+
+	  useEffect(()=>{
+		console.log(bookmarks+">>>>>>>>>>>>>>>>>>>>>>>>>.");
```

Comment:
Oh damn, have you re-written this hook? It already exists in `/screen/home.js`. Consider receiving those values(states) as props and update those rather than reinventing the wheel.

Reply:
Most of the `console.log`s are just for checking output. I'll remove them

@Shofiya2003 Do you want to take the second comment?

Reply:
@joshi-kaushal we have rewritten the use-effect because we too are fetching results of cities form api and there were no states of home.js we could have used as it is. If you could elaborate what change do you recommend

Reply:
> there were no states of home.js we could have used as it is.

I don't understand, could you please elaborate?

Reply:
I meant the states of home.js could not have been used as props for the Bookmark componenet